src/scandipwa/app/component/GoogleTagManager/events/Impression.event.js

```diff
@@ -59,39 +61,43 @@ class Impression extends BaseEvent {
      * Bind PWA event handling
      */
     bindEvent() {
-        // PLP
-        Event.observer(EVENT_GTM_IMPRESSIONS_PLP, ({ items, filters, category }) => {
-            this.handle(PLP_IMPRESSIONS, items, filters, category);
-        });
-
-        // Home
-        Event.observer(EVENT_GTM_IMPRESSIONS_HOME, ({ items, filters }) => {
-            this.handle(HOME_IMPRESSIONS, items, filters);
-        });
-
-        // Checkout Cross-sell
-        Event.observer(EVENT_GTM_IMPRESSIONS_CROSS_SELL, ({ items }) => {
-            this.handle(CHECKOUT_CROSS_SELL_IMPRESSIONS, items);
-        });
-
-        // Wishlist
-        Event.observer(EVENT_GTM_IMPRESSIONS_WISHLIST, ({ items }) => {
-            this.handle(WISHLIST_IMPRESSIONS, items);
-        });
-
-        // Search
-        Event.observer(EVENT_GTM_IMPRESSIONS_SEARCH, ({ items }) => {
-            this.handle(SEARCH_IMPRESSIONS, items);
-        });
-
-        // Recomended
-        Event.observer(EVENT_GTM_IMPRESSIONS_LINKED, ({ items }) => {
-            this.handle(RECOMMENDED_IMPRESSIONS, items);
-        });
-
-        Event.observer(EVENT_GTM_GENERAL_INIT, () => {
-            this.cleanStorage();
-        });
+        if (isEventEnabled(EVENT_IMPRESSION)) {
+            // PLP
+            Event.observer(EVENT_GTM_IMPRESSIONS_PLP, ({ items, filters, category }) => {
+                this.handle(PLP_IMPRESSIONS, items, filters, category);
+            });
+
+            // Home
+            Event.observer(EVENT_GTM_IMPRESSIONS_HOME, ({ items, filters }) => {
+                this.handle(HOME_IMPRESSIONS, items, filters);
+            });
+
+            // Checkout Cross-sell
+            Event.observer(EVENT_GTM_IMPRESSIONS_CROSS_SELL, ({ items }) => {
+                this.handle(CHECKOUT_CROSS_SELL_IMPRESSIONS, items);
+            });
+
+            // Wishlist
+            Event.observer(EVENT_GTM_IMPRESSIONS_WISHLIST, ({ items }) => {
+                this.handle(WISHLIST_IMPRESSIONS, items);
+            });
+
+            // Search
+            Event.observer(EVENT_GTM_IMPRESSIONS_SEARCH, ({ items }) => {
+                this.handle(SEARCH_IMPRESSIONS, items);
+            });
+
+            // Recomended
+            Event.observer(EVENT_GTM_IMPRESSIONS_LINKED, ({ items }) => {
+                this.handle(RECOMMENDED_IMPRESSIONS, items);
+            });
+        }
+
+        if (isEventEnabled(EVENT_GENERAL)) {
+            Event.observer(EVENT_GTM_GENERAL_INIT, () => {
+                this.cleanStorage();
+            });
+        }
```

Comment:
I'd split this into two separate functions and did `if (!cond) return;` in order to reduce nesting. Consider doing that during the logic refactor.